hack/go/generate_versions.go

```diff
@@ -181,7 +181,7 @@ func getVersion(repo, ref string) (string, error) {
 	baseURL := fmt.Sprintf("https://raw.githubusercontent.com/%s/%s", repo, ref)
 	link := fmt.Sprintf("%s/VERSION", baseURL)
 	if repo == metricsServerRepo {
-		link = fmt.Sprintf("%s/manifests/release/kustomization.yaml", baseURL)
+		link = fmt.Sprintf("%s/manifests/components/release/kustomization.yaml", baseURL)
```

Comment:
I'm afraid we'll have to branch as before `v0.7.0`, the file was in fact at `/manifests/release/kustomization.yaml`: https://github.com/kubernetes-sigs/metrics-server/blob/v0.6.4/manifests/release/kustomization.yaml

we started using v0.7.0 in 4.16, so we can branch on that.
Or fallback to the other path on a 404, as you want :)

Reply:
I didn't check, but maybe we could add a VERSION file in our downstream?
syncbot should know the tag, we could just ask to put it there...

Reply:
Ah! [That](https://github.com/openshift/cluster-monitoring-operator/pull/2252/files#r1474205497) makes so much sense, I missed this! This will definitely break jobs on `<4.16` PRs, true. [This](https://github.com/openshift/cluster-monitoring-operator/pull/2252/files#r1474211363) sounds like a good long-term approach, but would adding a conditional for checking both paths suffice here?

Reply:
Yes let's branch to unblock the CI, we'll try the other approach later.

Reply:
Just to confirm, if https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-monitoring-operator/openshift-cluster-monitoring-operator-release-4.16.yaml and https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-monitoring-operator/openshift-cluster-monitoring-operator-release-4.15.yaml are sourced from the respective branches instead of `master`, wouldn't that automatically fix things for us?

Reply:
(since this commit won't go into `4.15` as it's out, and only `4.16` will be up-to-date with `master`?)

Reply:
But with the code from `master` we fetch versions for `4.14` and `4.17`, so it should be compatible with both.

Reply:
Right, I was confused by that. Could you elaborate on the branching approach, since we already have branches till `4.17` on both. I'd also like to reiterate the idea of having a conditional that checks both `/manifests/release/kustomization.yaml` and `/manifests/components/release/kustomization.yaml` for a `200`, and goes with that, which would fix things on both fronts. We can drop the older path once the CMO versions using them correspond to EOL releases. WDYT?

Reply:
Sorry, I wasn't explicit, by branching I mean "conditional branching": adding a condition on the version